src/test/ui-test/install.test.ts

```diff
@@ -0,0 +1,238 @@
+        const [_, extension] = await openExtensionPage(marketplace, name, this.timeout() - 1000);
+        expect(await extension.getDescription()).to.be.equal(extensionMetadata['description']);
+    });
+    it('Has correct author', async function () {
+        const [_, extension] = await openExtensionPage(marketplace, name, this.timeout() - 1000);
+        expect(await extension.getAuthor()).to.be.equal('Red Hat');
+    });
+    it('Extension is installed', async function () {
+        const [_, extension] = await openExtensionPage(marketplace, name, this.timeout() - 1000);
+        expect(await extension.isInstalled()).to.be.true;
+    });
+    it('All commands are available', async function () {
+        for (const commandMetadata of extensionMetadata['contributes']['commands']) {
+            const { command, title } = commandMetadata;
+            let input = await getInput();
+            await input.setText(`>${title}`);
+            await repeat(async () => {
+                try {
+                    const quickpick = await input.findQuickPick(title);
+                    return await quickpick?.getLabel() === title;
+                }
+                catch (e) {
+                    // Input cannot be stale by design. Refresh quickpicks.
+                    if (e instanceof error.StaleElementReferenceError) {
+                        return undefined;
+                    }
+                    // Some element could not be found. Try again.
+                    if (e instanceof error.NoSuchElementError) {
+                        return undefined;
+                    }
+                    // Input was unexpectedly closed. Open it again.
+                    if (e instanceof error.ElementNotVisibleError) {
+                        input = await getInput();
+                        await input.setText(`>${title}`);
+                        return undefined;
+                    }
+
+                    throw e;
+                }
+            }, {
+                id: `Verify command '${command}'.`,
+                timeout: this.timeout() - 1000,
+                message: `Could not find commnad '${command}' in quickpicks.`
+            });
+        }
+    });
+});
+
+/**
+ * Open input and return its reference when it is ready.
+ * @returns Input ready to be used.
+ */
+async function getInput(): Promise<QuickOpenBox> {
+    const workbench = new Workbench();
+    const input = await workbench.openCommandPrompt();
+    return input.wait();
+}
+
+/**
+ * Open the extension page.
+ * @param marketplace Reference to ExtensionViewSection.
+ * @param name Display name of the extension.
+ * @param timeout Timeout in ms.
+ * @returns A tuple -- updated ExtensionViewSection if it has became stale and ExtensionViewItem
+ *          object tied with the extension.
+ */
+async function openExtensionPage(marketplace: ExtensionsViewSection, name: string, timeout: number): Promise<[ExtensionsViewSection, ExtensionsViewItem]> {
```

Comment:
It would be ideal. The problem is these methods change a lot and would need to be republished to npm on every patch. But for now I would leave it as it is.

Reply:
Regarding theia-extension-tester dependencies, it is just collection of packages which were developed for theia-extension-tester. However the most of them are modular and reusable in other projects. The repeat package was dependent on extension-tester-page-objects but thanks to your feedback I have removed the page-objects dependency from the repeat package.